libs/ledger/src/chain/main_chain.cpp

```diff
@@ -2082,11 +2082,18 @@ DigestSet MainChain::DetectDuplicateTransactions(BlockHash const &           sta
 
   for (auto const &tx_layout : transactions)
   {
+    auto const valid_until = tx_layout.valid_until();
+
     auto const default_valid_from =
-        tx_layout.valid_until() - std::min(MAXIMUM_TX_VALIDITY_PERIOD, tx_layout.valid_until());
+        valid_until >= chain::Transaction::MAXIMUM_TX_VALIDITY_PERIOD
+            ? (valid_until - chain::Transaction::MAXIMUM_TX_VALIDITY_PERIOD)
+            : 0;
+
+    static_assert(chain::Transaction::MAXIMUM_TX_VALIDITY_PERIOD == 40000, "");
```

Comment:
I am not sure that I would include this in the review. This was only for our local testing I thought?

Reply:
OK, I will remove it